decompiler/main.cpp

```diff
@@ -4,16 +4,23 @@
 #include "ObjectFile/ObjectFileDB.h"
 #include "config.h"
 #include "util/FileIO.h"
-
+#include "third-party/spdlog/include/spdlog/spdlog.h"
+#include "third-party/spdlog/include/spdlog/sinks/basic_file_sink.h"
 #include "common/util/FileUtil.h"
 
 int main(int argc, char** argv) {
-  printf("Jak Disassembler\n");
+  spdlog::info("Beginning disassembly. This may take a few minutes...");
+
+  spdlog::set_level(spdlog::level::debug);
+  auto lu = spdlog::basic_logger_mt("GOAL Decompiler", "logs/decompiler.log");
+  spdlog::set_default_logger(lu);
+  spdlog::flush_on(spdlog::level::info);
+
   init_crc();
   init_opcode_info();
 
   if (argc != 4) {
-    printf("usage: jak_disassembler <config_file> <in_folder> <out_folder>\n");
+    spdlog::info("Usage: jak_disassembler <config_file> <in_folder> <out_folder>");
```

Comment:
Let's leave this one as a printf - this is an error message that shows up if you get the arguments wrong which would be nice to see in the terminal.